src/components/ArticleActions.vue

```diff
@@ -95,6 +99,21 @@ export default {
       } catch (err) {
         console.error(err);
       }
+    },
+    async cloneArticle() {
+      try {
+        await this.$store.dispatch(CLONE_ARTICLE, this.article.slug);
+        this.$router.push("/");
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    async toggleVisibility() {
+      if (this.article.visible) {
+        makeArticleDisappear(this.article);
+      } else {
+        makeArticleAppear(this.article);
+      }
```

Comment:
nit: you could just use a ternary operator here

```suggestion
      this.article.visible ? makeArticleDisappear(this.article) : makeArticleAppear(this.article);
```